vispy/scene/node.py

```diff
@@ -626,3 +627,15 @@ def picking(self, p):
         for c in self.children:
             c.picking = p
         self._picking = p
+
+    @contextmanager
+    def picking_context(self, *, picking=True):
+        """Context manager to temporarily set picking for this node and its
+        children.
+        """
```

Comment:
```suggestion
    def set_picking(self, *, picking=True):
        """Context manager to temporarily set picking for this node and its children."""
```

The docstring "subject" should be one line. We don't have any linters enabled that complain about this right now, but I'd like to add them in the future. If this causes it to be a really long line then maybe we could shorten the description and put the rest into a "body".

Reply:
Otherwise, I'm wondering about a different name still. Not sure `set_picking` is descriptive enough or clear enough that it should be used as a context manager.

Reply:
I'm not sure of a convention to indicate something is a context manager but `set_picking` sounds too much like a regular setter to me. Maybe `picking_mode`? We could make it more explicit with something like `visual_picking` or `pick_visuals` - this would also differentiate from the primitive picking filters proposed in #2500
(which maybe should also offer context managers...).

Reply:
I think `set_picking` could be good if we also make it work as a normal setter method (much like the gl_state methods).

Reply:
I am OK with that. :+1: